crates/runtime/src/actor/component/mod.rs

```diff
@@ -37,16 +37,16 @@ mod messaging;
 pub(crate) use self::http::incoming_http_bindings;
 pub(crate) use self::logging::logging_bindings;
 
-type TableResult<T> = Result<T, TableError>;
+type TableResult<T> = Result<T, ResourceTableError>;
 
 mod guest_bindings {
     wasmtime::component::bindgen!({
         world: "guest",
         async: true,
         with: {
-           "wasi:io/error@0.2.0-rc-2023-11-10": wasmtime_wasi::preview2::bindings::io::error,
-           "wasi:io/poll@0.2.0-rc-2023-11-10": wasmtime_wasi::preview2::bindings::io::poll,
-           "wasi:io/streams@0.2.0-rc-2023-11-10": wasmtime_wasi::preview2::bindings::io::streams,
```

Comment:
I saw this change in a few places, do we not need to version anymore because the bindings are under `wasmtime_wasi::preview2`?

I want to understand when there is a bump, e.g. 0.2.1, will we need to version that specifically or will there be a different binding

Reply:
We simply don't need the version - the host only ever implement one version anyway. That's how this macro is used in Wasmtime, honestly I don't know about the specifics.
We will most probably not need to change anything for 0.2.1, but if we do, we will just do like is done in Wasmtime

Reply:
I'm thinking forward to WASI p3 where we have components that work with and are built for 0.2.0 but we still want to support them, I assume most APIs may bump versions but remain the same, so I just wanted to see how that will work at the bindings level. Going to resolve this conversation as it's not something that should block this PR